recovery/systemrescue/sas-recovery.sh

```
@@ -0,0 +1,487 @@
+      --mode) mode=${2:-}; shift 2;;
+      --max-chars) max=${2:-}; shift 2;;
+      *) die "unknown qr-catalog argument: $1";;
+    esac
+  done
+  [ -n "$repo_mount" ] && [ -n "$source" ] && [ -n "$destination" ] && [ -n "$workdir" ] && [ -n "$image" ] && [ -n "$map" ] || die "qr-catalog requires --repo-mount, --source, --destination, --workdir, --image, and --map"
+  case "$mode" in start|resume) ;; *) die "--mode must be start or resume";; esac
+  [[ "$max" =~ ^[0-9]+$ ]] || die "--max-chars must be numeric"
+  local runner="$repo_mount/recovery/systemrescue/sas-recovery.sh"
+  local state="export R='$runner' SRC='$source' DST='$destination' M='$dest_mount' W='$workdir' IMG='$image' MAP='$map'"
+  emit_qr 1 "$state" "$max"
+  emit_qr 2 'bash "$R" protect-source --source "$SRC"' "$max"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)